src/main/java/io/strimzi/kafka/bridge/SourceBridgeEndpoint.java

```diff
@@ -80,11 +89,33 @@ protected void handleClose() {
      */
     protected void send(KafkaProducerRecord<String, byte[]> krecord, Handler<AsyncResult<RecordMetadata>> handler) {
 
-        log.debug("Sending to Kafka on topic {} at partition {} and key {}", krecord.topic(), krecord.partition(), krecord.key());
-        if (handler == null) {
-            this.producerSettledMode.write(krecord);
-        } else {
-            this.producerUnsettledMode.write(krecord, handler);
+        Properties props = new Properties();
+        KafkaConfig consumerConfig = this.bridgeConfigProperties.getKafkaConfig();
+        props.put("bootstrap.servers", consumerConfig.getBootstrapServers());
+        AdminClient ac = AdminClient.create(props);
```

Comment:
not sure that we need the `AdminClient` which checks stuff before sending ... this will drive a slower client experience I guess. You should check what kind of exceptions are raised just sending a message when the above conditions happen (topic or partition don't exist). Finally, I am not sure it's the right place to do this check or at the upper level. Remember that this part is shared with the AMQP bridge so it should be agnostic. But we have to think more about that in terms of architecture.